crates/era-cheatcodes/src/cheatcodes.rs

```diff
@@ -2073,60 +2117,77 @@ impl CheatcodeTracer {
             return
         }
 
-        // Skip check if opcode is not Ret
```

Comment:
This might need to be reintroduced here (also to avoid right drift) and removed below

Reply:
or perhaps we should consider renaming `ActionOnReturn` and `handle_return` since it's not only for that anymore...

Reply:
Moved up ret opcode check. Handle_return makes sense since it will only trigger on ret opcodes, at least at the moment.